app/javascript/packages/step-indicator/step-indicator.tsx

```diff
@@ -0,0 +1,37 @@
+import type { HTMLAttributes, ReactNode } from 'react';
+import { t } from '@18f/identity-i18n';
+import StepIndicatorElement from './step-indicator-element';
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'lg-step-indicator': HTMLAttributes<StepIndicatorElement> & { class?: string };
+    }
+  }
+}
+
+interface StepIndicatorProps {
+  /**
+   * Extra CSS classes to apply to wrapper.
+   */
+  className?: string;
+
+  /**
+   * Step indicator children, expected to be provided as StepIndicatorStep elements.
+   */
+  children: ReactNode;
+}
+
+function StepIndicator({ className, children }: StepIndicatorProps) {
+  return (
+    <lg-step-indicator
+      role="region"
+      aria-label={t('step_indicator.accessible_label')}
+      class={className}
```

Comment:
Bizarrely, React requires class names to be applied as `class` for custom elements, not `className`:

>One common confusion is that Web Components use “class” instead of “className”.

https://reactjs.org/docs/web-components.html#using-web-components-in-react

Bizarre especially because `className` makes sense if you think about it being reflective of the [`Element#className`](https://developer.mozilla.org/en-US/docs/Web/API/Element/className) property being controlled, which is still a property which would exist on a custom element 🤷 

Related: https://github.com/facebook/react/issues/4933

Reply:
let's switch to preact so we can just use `class` everywhere and not sweat it

Reply:
> let's switch to preact so we can just use `class` everywhere and not sweat it

Can't tell if serious 😅

Reply:
25% serious, I don't have a sense of how much effort a real rewrite would be

Reply:
> 25% serious, I don't have a sense of how much effort a real rewrite would be

As much as I love Preact, I'd still have some reservations about migrating, even if straightforward, due to losing some of the ecosystem maturity, knowledge/familiarity implications around hiring, and overall developer experience.